lib/get-config.js

```diff
@@ -303,7 +310,60 @@ function validateRules(config, options) {
   }
 }
 
-module.exports = function(options) {
+function _determineConfigForSeverity(config) {
+  switch (config) {
+    case SEVERITY_LEVEL_IGNORE:
+      return { config: false, severity: IGNORE_SEVERITY };
+    case SEVERITY_LEVEL_WARNING:
+      return { config: true, severity: WARNING_SEVERITY };
+    case SEVERITY_LEVEL_ERROR:
+      return { config: true, severity: ERROR_SEVERITY };
+  }
+}
+
+/**
+ * Sets the severity and config on the rule object.
+ * Eg:
+ * {'no-implicit-this': 'warn'} -> {'no-implicit-this': { severity: 'warn', config: true }}
+ * { 'no-implicit-this': [ 'warn', { allow: [ 'fooData' ] } ] }
+ * would become:
+ * { 'no-implicit-this': { severity: 'warn', config: 'allow': [ 'fooData' ] } }
+ * @param {*} configData
+ */
+function processRules(config) {
+  for (let key in config.rules) {
+    let severityConfig;
+    let ruleData = config.rules[key];
+    let ruleConfig = {};
+    // In case of {'no-implicit-this': 'off|warn|error'}
+    if (typeof ruleData === 'string') {
+      severityConfig = _determineConfigForSeverity(ruleData);
+      if (severityConfig) {
+        ruleConfig.severity = severityConfig.severity;
+        ruleConfig.config = severityConfig.config;
+        config.rules[key] = ruleConfig;
+      }
+    }
+    // array of severity and custom rule config
+    else if (Array.isArray(ruleData)) {
+      if (ruleData.length !== 2) {
+        throw new Error(
+          'Severity and custom rule config are the only two values allowed in the rule config array.'
+        );
+      }
+      let severity = ruleData[0];
+      severityConfig = _determineConfigForSeverity(severity);
+      if (severityConfig === undefined) {
+        throw new Error('Severity can only be "off", "warn", or "error"');
+      }
```

Comment:
Same thing as above, arrays are currently valid as rule config. We cannot throw if we see an array like `no-invalid-tagnames: ['div', 'table'],`.